### ISSUE TITLE
add pure change mode

### ISSUE THREAD

**Issue by shaunlebron** [parinfer/parinfer.js#75]
Due to the recent performance enhancements (see below), I believe we can now create an optimized change function that doesn't require state.  This will free Atom (and other editors) from manual performance management without introducing complexity of a stateful api.

```
Processing really_long_file : 2865 lines, 112431 chars
read: 5.801ms
indent: 47.229ms
paren: 41.816ms
```

These numbers will result in < 20ms to process a change, given the following steps:
- read the entire file to get the lines of the top-level open-parens (takes ~5ms)
- only process the top-level expressions that intersect the lines that were changed (takes ~11ms for 300 lines)

working on the api

**Comment by shaunlebron** [comment]
Turns out that state still has to be tracked.  All edits between debounced parinfer calls must be aggregated.  This means edited line numbers must be remembered.  This is further complicated by having to update currently aggregated line numbers when lines are inserted are removed before them.

Further still, edits must be aggregated across _unsuccessful_ parinfer calls (i.e. those returning success = false), so that top-level expressions from previous edits don't get stale.

In conclusion, I'm willing to sacrifice ~30ms to keep the API simple (i.e. by not requiring API consumers to track state).  Closing.
